src/providers/sambanova/chatComplete.ts

```diff
@@ -0,0 +1,86 @@
+import { SAMBANOVA } from '../../globals';
+import {
+  ChatCompletionResponse,
+  ErrorResponse,
+  ProviderConfig,
+} from '../types';
+
+export interface SamvaNovaChatCompleteResponse extends ChatCompletionResponse {}
+
+export interface SamvaNovaErrorResponse extends ErrorResponse {}
+
+export interface SamvaNovaStreamChunk {
+  id: string;
+  object: string;
+  created: number;
+  model: string;
+  system_fingerprint: string;
+  choices: {
+    delta: {
+      content?: string;
+    };
+    index: number;
+    finish_reason: string | null;
+    logprobs: object | null;
+  }[];
+  usage: {
+    is_last_response: boolean;
+    total_tokens: number;
+    prompt_tokens: number;
+    completion_tokens: number;
+    time_to_first_token: number;
+    end_time: number;
+    start_time: number;
+    total_latency: number;
+    total_tokens_per_sec: number;
+    completion_tokens_per_sec: number;
+    completion_tokens_after_first_per_sec: number;
+    completion_tokens_after_first_per_sec_first_ten: number;
+  } | null;
+}
+
+export const SamvaNovaChatCompleteStreamChunkTransform: (
+  response: string
+) => string = (responseChunk) => {
+  let chunk = responseChunk.trim();
+  chunk = chunk.replace(/^data: /, '');
+  chunk = chunk.trim();
+  if (chunk === '[DONE]') {
+    return `data: ${chunk}\n\n`;
+  }
+
+  const parsedChunk: SamvaNovaStreamChunk = JSON.parse(chunk);
+  if (parsedChunk.usage) {
+    return `data: ${JSON.stringify({
+      id: parsedChunk.id,
+      object: parsedChunk.object,
+      created: parsedChunk.created,
+      model: parsedChunk.model,
+      provider: SAMBANOVA,
+      choices: [],
+      usage: {
+        prompt_tokens: parsedChunk.usage.prompt_tokens || 0,
+        completion_tokens: parsedChunk.usage.completion_tokens || 0,
+        total_tokens: parsedChunk.usage.total_tokens || 0,
+      },
+    })}\n\n`;
+  }
+  return `data: ${JSON.stringify({
+    id: parsedChunk.id,
+    object: parsedChunk.object,
+    created: parsedChunk.created,
+    model: parsedChunk.model,
+    provider: SAMBANOVA,
+    choices: [
```

Comment:
By the way, are there choices array always/should return single item? looks like it should do like
```ts
choices: parsedChunk.choices.map(choice =>{ ...
```
, but i see other providers doing like this.

Reply:
You can review this example response for `cohere` provider.

```
data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"I am"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":" pow"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"ered"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":" by "}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"Comm"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"and,"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":" a l"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"arge"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":" lan"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"guag"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"e mo"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"del "}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"buil"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"t by"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":" the"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":" com"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"pany"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":" Coh"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"ere."}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":" Coh"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"ere "}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{"role":"assistant","content":"is"}}]}

data: {"id":"e253ffbf-ebd0-4bc1-8cbf-37d597625500","object":"chat.completion.chunk","created":1725344920657,"model":"","system_fingerprint":null,"provider":"cohere","usage":{"completion_tokens":20,"prompt_tokens":17,"total_tokens":37},"choices":[{"index":0,"delta":{},"finish_reason":"MAX_TOKENS"}]}

data: [DONE]

```